### ISSUE TITLE
HttpApp - should it also mix-in Directives

### ISSUE THREAD

**Issue by ktoso** [akka/akka-http#875]
This I'm actually not sure I'd want but wanted to discuss how you feel about it @jlprat... 

To be the "one and single extends you need" we could also make it extend Directives

Nowadays to be ready to work with it I'd need:

```
object FeedFromHttpPostApp extends App with HttpApp with Directives
```

WDYT?

**Comment by jlprat** [comment]
I'm in for it, didn't do it before because mainly all Scala snippets do not extend `Directives`, but import them. I wanted to avoid the feeling that "`HttpApp`  is deciding too many things for me"

You know it was a long PR ;)

**Comment by jlprat** [comment]
BTW, the Java version extends `AllDirectives`. So I guess it makes sense to do the same for the Scala one

**Comment by ktoso** [comment]
Hm yeah we should make them 1:1 matching.

Looking for down sides but so far seems ok to do.

**Comment by jlprat** [comment]
I can add this to the https://github.com/akka/akka-http/pull/876

**Comment by jrudolph** [comment]
Fixed by #876.
